setup/terraform/modules/aks/main.tf

```diff
@@ -15,15 +15,18 @@ resource "azurerm_kubernetes_cluster" "aks" {
   kubernetes_version  = "1.29.7"
   dns_prefix          = var.aks_cluster_name
 
+  oidc_issuer_enabled       = true # OIDC issuer is enabled for AKS cluster authentication with Azure AD
+  workload_identity_enabled = true # Workload identity allows the AKS cluster to use managed identities for Azure resources
+
   # AC-22, IA-8, SC-2, SC-5: Configure the private cluster settings
   private_cluster_enabled = var.use_private_network
-  private_dns_zone_id     = var.use_private_network ? "System" : null
 
   # Configure the default node pool
   default_node_pool {
-    name       = "default"
-    node_count = 2
-    vm_size    = "Standard_D4s_v3"
+    name           = "default"
+    node_count     = 2
```

Comment:
If we have pods autoscaling, don't we also need this to autoscale? (Or have more nodes)
I realize this may be bound by organizational limits on vCPUs though.

Reply:
I've added a prerequisite that DEV's quota limit extend to 16 and other environments go to 64. Then I added an environment variable for this amount and we adjust the max node count based on that and the chosen size. With 64 vCPU and a node size with 4 vCPUs, the max node count would be computed as 16.